DropboxLike.Api/Controllers/FileController.cs

```diff
@@ -19,11 +19,17 @@ public FileController(IFileService fileService)
   [Route("Upload")]
   public async Task<IActionResult> UploadFileAsync(IFormFile file)
   {
-    var fileExt = Path.GetExtension(file.Name);
-    var objName = $"{Guid.NewGuid()}.{fileExt}";
-
-    var response = _fileService.UploadSingleFile(file);
+    var response = await _fileService.UploadSingleFileAsync(file);
 
     return Ok(response);
   }
+
+  [HttpGet]
+  [Route("Download")]
+  public async Task<IActionResult> DownloadFileAsync(string request)
```

Comment:
What I would do is store metadata about the files users have stored in AWS in a relational database. That means you would have one endpoint that allows you to retrieve metadata about all the files a user has stored, with properties such as file ID (used in database), file location (reference to some folder if you want to implement folders in the end), file name, file extension, creation date, and modification date, the latter five of which would be used to render the file in the frontend.
This endpoint would then become quite easy: in order for a user to download a file that is rendered in the frontend from the metadata you retrieved earlier (*which includes the ID!!*), when they click the button, you send a request to `/download/${ID}`. You can pass this ID on to your service which will download the file by checking if metadata about this file exists in the database and then proceeds to build a request to AWS S3 based on the metadata about this file.
This is actually similar to what I implemented myself back when I did this project.

Reply:
If you want to fix the issue you are having right now, try instead to create a class `DownloadFileRequest` with property `fileName` of type `string`, and then pass a JSON body in Postman that looks like
```
{
    "fileName": "someFileName.pdf"
}
```
This should allow you to debug your current implementation.

Reply:
I just looked up some resources, one that might be useful is as follows.
https://learn.microsoft.com/en-us/aspnet/web-api/overview/formats-and-model-binding/parameter-binding-in-aspnet-web-api
As a matter of fact, that means adding the attribute `[FromBody]` and then sending the same JSON payload mentioned in my previous comment as the request body should also work.